lib/types/app.types.ts

```diff
@@ -25,10 +25,10 @@ export interface AppConfigAPI {
   getParameters: <T extends KeyValueMap = KeyValueMap>() => Promise<null | T>
   /** Registers a handler to be called to produce parameters for an App */
   onConfigure: (
-    handler: () => {
-      parameters?: KeyValueMap
-      targetState?: AppState
-    }
+    handler: () =>
+      | Promise<{ parameters?: KeyValueMap; targetState?: AppState }>
+      | { parameters?: KeyValueMap; targetState?: AppState }
```

Comment:
makes sense to make it it's own type and reuse it here and the other file. Considering it's even explained in docs

Reply:
makes sense. Also going to export it to simplify typing the actual app